### PR TITLE
doc: TF-M limitations update in TF-M ug

### PR DESCRIPTION
**Pull request by mia-ko**
Updated the Limitations section in TF-M ug according to NCSDK-20888.
Fixed missing prefix for two TF-M samples.

### CODE REVIEW

File: doc/nrf/app_dev/tfm/index.rst [hunk 1]
```diff
@@ -129,25 +129,17 @@ There are several options to get UART output from the secure TF-M:
...
-
-In Isolation level 2 or higher, the number of peripherals configured as secure in Application Root of Trust (ARoT) is limited by the number of available MPU regions.
-
-Nordic platforms support only the GCC toolchain for building TF-M.
-
-Enabling Floating point support in TF-M is currently not supported.
-Enabling Floating point Hard ABI (:kconfig:option:`CONFIG_FP_HARDABI`) in the application is currently not supported.
+* Isolation level 3 is not supported.
+* In Isolation level 2 or higher, the number of peripherals configured as secure in Application Root of Trust (ARoT) is limited by the number of available MPU regions.
+* Nordic platforms support only the GCC toolchain for building TF-M.
```
Comment (mia-ko) [inline doc/nrf/app_dev/tfm/index.rst:142]: Not sure about the Nordic platforms here... Nordic Semiconductor platforms? Or is nRF Connect SDK platforms more accurate?

File: samples/tfm/provisioning_image/README.rst [hunk 2]
```diff
@@ -1,7 +1,7 @@
 .. _provisioning_image:
 
-Provisioning image
-##################
+TF-M: Provisioning image
```
Comment (mia-ko) [inline samples/tfm/provisioning_image/README.rst:3]: I know these prefix updates are not directly related to the other update, it was just convenient to use the same branch to fix when it was pointed out by the release team.

File: doc/nrf/app_dev/tfm/index.rst [hunk 3]
```diff
@@ -129,25 +129,17 @@ There are several options to get UART output from the secure TF-M:
...
 
-* Audit Log service in IPC model
-* Firmware Update service
-* Attestation service
+* Firmware Update service is not supported.
+* The following crypto modules or ciphers are not supported:
 
-The following Crypto modules or ciphers are not supported:
+  * Output feedback (OFB) mode.
+  * Cipher feedback (CFB) mode.
```
Comment (joerchan) [inline doc/nrf/app_dev/tfm/index.rst:138]: AES output feedback (AES-OFB) mode.
AES cipher feedback (AES-CFB) mode.

File: samples/tfm/provisioning_image/README.rst (hunk 2 above)
Comment (joerchan) [reply samples/tfm/provisioning_image/README.rst:3]: That's fine, you could use a separate commit to show that they are not related (not important).

File: doc/nrf/app_dev/tfm/index.rst (hunk 1 above)
Comment (joerchan) [reply doc/nrf/app_dev/tfm/index.rst:142]: Whatever you want to say to refer to our SoCs.

File: samples/tfm/provisioning_image/README.rst (hunk 2 above)
Comment (mia-ko) [reply samples/tfm/provisioning_image/README.rst:3]: Yes, that would have been a better way of doing it... :)
Ok, I'll keep it the way it is this time, but will keep in mind having separate commits next time I do something similar.

File: doc/nrf/app_dev/tfm/index.rst (hunk 1 above)
Comment (mia-ko) [reply doc/nrf/app_dev/tfm/index.rst:142]: I would have said Nordic Semiconductor SoCs. But then, nRF9160 is not an SoC, it's a SiP. So probably Nordic Semiconductor's hardware platforms then.
Are we ok with that?

Comment (joerchan) [reply doc/nrf/app_dev/tfm/index.rst:142]: There must be an already established way to refer to this?

`Nordic Semiconductor devices`?

Comment (mia-ko) [reply doc/nrf/app_dev/tfm/index.rst:142]: Yes, that's the second way of writing it. I only found these two that we have mentioned now. So we go for your version then :)

**Review by joerchan — APPROVED** [review]

File: samples/tfm/provisioning_image/README.rst (hunk 2 above)
Comment (melwee01) [reply samples/tfm/provisioning_image/README.rst:3]: Never mind, I was misreading it!

**Review by melwee01 — APPROVED** [review]
